### PR TITLE
TRC: EventGroup hashing

### PR DESCRIPTION
**Pull request by robem**
This PR copies the hashing implementation from the TRS and adds two tests. Please see commit message for detail.

### CODE REVIEW

File: client/thin-replica-client/src/trc_hash.cpp [hunk 1]
```diff
@@ -68,40 +70,80 @@ static string hashUpdateFromEntryHashes(uint64_t block_id, const map<string, str
...
+
+    return hashUpdateFromEntryHashes(legacy_event.block_id, entry_hashes);
   }
 
-  return hashUpdateFromEntryHashes(legacy_event.block_id, entry_hashes);
+  ConcordAssert(std::holds_alternative<EventGroup>(ev));
+  auto& event_group = std::get<EventGroup>(ev);
+  set<string> entry_hashes;
+  for (const auto& event : event_group.events) {
+    string event_hash = ComputeSHA256Hash(event);
```
Comment (shgandhi) [inline client/thin-replica-client/src/trc_hash.cpp:117]: Event includes tags as well, we don't want them to be included in the hash afaik, and only hash `event.data`.

File: client/thin-replica-client/src/trc_hash.cpp [hunk 2]
```diff
@@ -68,40 +70,80 @@ static string hashUpdateFromEntryHashes(uint64_t block_id, const map<string, str
...
+      entry_hashes[key_hash] = ComputeSHA256Hash(kvp.value());
     }
-    entry_hashes[key_hash] = ComputeSHA256Hash(kvp.value());
+
+    return hashUpdateFromEntryHashes(update.events().block_id(), entry_hashes);
+  }
+  ConcordAssert(update.has_event_group());
+  set<string> entry_hashes;
+  for (const auto& event : update.event_group().events()) {
+    string event_hash = ComputeSHA256Hash(event);
```
Comment (shgandhi) [inline client/thin-replica-client/src/trc_hash.cpp:141]: Same comment as above, here.

**Review by shgandhi — APPROVED** [review]
Thanks for the change @robem, LGTM! I have one comment though, if you could take a look.

File: client/thin-replica-client/src/trc_hash.cpp (hunk 1 above)
Comment (robem) [reply client/thin-replica-client/src/trc_hash.cpp:117]: The `event` in this for-loop comes from EventGroup.events which is a `vector<string>` type. If we include tags on the server side then we need to remove them there before sending.

File: client/thin-replica-client/src/trc_hash.cpp (hunk 2 above)
Comment (robem) [reply client/thin-replica-client/src/trc_hash.cpp:141]: The event in this for-loop comes from EventGroup.events (thin_replica.proto) which is a repeated bytes field so essentially a `vector<string>` type. Tags shouldn't be included by the server.

**Pull request merged**